File: Python/8py 1/combat.py
"""
|--------------------------------------------------------------------------
| Grasshopper - Terminal game combat function
|--------------------------------------------------------------------------
|
"""

def combat(health, damage):
   return max(0, health-damage)


def combat(health, damage):
   return health - damage if health > damage else 0


def combat(h, d):
   return (h>d)*(h-d)


def combat(health, damage):
   return [0, health-damage][damage < health]


print(combat(100, 5), 95)
print(combat(83, 16), 67)
print(combat(20, 30), 0)

File: PHP/8php 1/combat.php
<?php
/*
|--------------------------------------------------------------------------
| Grasshopper - Terminal game combat function
|--------------------------------------------------------------------------
|
*/

function combat($health, $damage) {
   return max(0, $health-$damage);
}


function combat($h, $d) {
   return $h < $d ? 0 : $h - $d;
}


function combat($h, $d) {
   return ($h -= $d) < 0 ? 0 : $h;
}


print_r(combat(100, 5) . ' ' . 95 . '
');
print_r(combat(92, 8) . ' ' . 84 . '
');
print_r(combat(20, 30) . ' ' . 0 . '
');